File: elixir/exam1_challenge.exs
defmodule Exam1 do
  def hello_message(name) do
    template_messages = [
    "Hello, #{name}, nice to meet you!",
    "Hi, #{name}, nice to meet you!"]

    result = Enum.take_random(template_messages, 1)
    IO.puts result
  end

  def start do
    name = String.replace_trailing((IO.gets "What is your name? "), "\n", "")
    hello_message(name)
  end
end

Exam1.start

File: ruby/exam1_challenge.rb
def hello_message(name)
  message_template = [
    "Hello, {name}, nice to meet you",
    "Hi, {name}, to nice to meet you"
  ]

  message = message_template[rand(message_template.size)]
  puts message.gsub(/{name}/, name)
end

print "What is your name? "

hello_message(gets.chop)


